lib/helper/transform.go

```diff
@@ -130,3 +130,100 @@ func FindCARByName(cars []kion.CAR, carName string) (*kion.CAR, error) {
 	}
 	return &kion.CAR{}, fmt.Errorf("cannot find cloud access role with name %v", carName)
 }
+
+// NormalizeAccessType normalizes the access type string returned from the Kion API
+// to the values used by the CLI.
+// It converts "console_access" to "web", and
+// "short_term_key_access" to "cli".
+// If the access type does not match any of these, it returns the original string.
+func NormalizeAccessType(accessType string) string {
+	switch accessType {
+	case "console_access":
+		return "web"
+	case "short_term_key_access":
+		return "cli"
+	default:
+		return accessType
+	}
+}
+
+// CombineFavorites combines local favorites with API favorites, ensuring that
+// local favorites are prioritized and that there are no duplicates or name conflicts.
+// It returns a slice of Favorites that contains all unique favorites, with local
+// favorites appearing first. If an API favorite has no name, it attempts to
+// generate a name based on the account name, car, access type, and region.
+func CombineFavorites(localFavs []structs.Favorite, apiFavs []structs.Favorite) (structs.FavoritesComparison, error) {
+	result := structs.FavoritesComparison{
+		All:        []structs.Favorite{},
+		Exact:      []structs.Favorite{},
+		NonMatches: []structs.Favorite{},
+		Conflicts:  []structs.Favorite{},
+		LocalOnly:  []structs.Favorite{},
+	}
+
+	// Track all "exact" matches by a unique composite key
+	exactMap := make(map[string]bool)
+
+	// Start with local favorites in the final set
+	result.All = append(result.All, localFavs...)
+
+	for _, apiFav := range apiFavs {
+
+		// default region to us-east-1
+		if apiFav.Region == "" {
+			apiFav.Region = "us-east-1"
```

Comment:
Should we make this configurable?

Reply:
Yes, that'd be better. Does the config file allow for setting the default region?

Reply:
Adde a DefaultRegion to the Kion struct and am referencing that in this function now.